tests/cpp/multidevice_transformer.h

```diff
@@ -0,0 +1,89 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION &
+ * AFFILIATES. All rights reserved. SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#pragma once
+
+#include <stdint.h>
+
+#include <runtime/fusion_executor_cache.h>
+
+namespace nvfuser {
```

Comment:
This is just copy and pasted from tests/cpp/multidevice_transformer.cpp
No need to read the actual code, aside from the class structure.